tools/witx/src/parser.rs

```diff
@@ -200,12 +281,14 @@ pub enum TopLevelSyntax<'a> {
 
 impl<'a> Parse<'a> for TopLevelSyntax<'a> {
     fn parse(parser: Parser<'a>) -> Result<Self> {
-        if parser.peek::<kw::r#use>() {
-            parser.parse::<kw::r#use>()?;
-            Ok(TopLevelSyntax::Use(parser.parse()?))
-        } else {
-            Ok(TopLevelSyntax::Decl(parser.parse()?))
-        }
+        parser.parens(|p| {
```

Comment:
It might be a bit easier if the `parser.parens` were moved directly inside `Documented` perhaps? Or maybe something like `Documented` and then like `DocumentedIdent` for enums/flags/etc

Reply:
The parens combinator was inside `Documented` but I had to move it out. Some items that need to be Documented are ordinary identifiers (enum & flag variants) not surrounded by parens.